lib/server.coffee

```diff
@@ -111,7 +113,6 @@ exports.request = (options = {}, outerCallback, onProgress) ->
 				progress(request(options))
 					.on('progress', onProgress)
 					.on('error', outerCallback)
-					.on('end', onProgress)
```

Comment:
@Page- I've did some testing, at `close` handles all cases. Having the `end` listener call `onProgress` results in `onProgress` having a `null` state argument, and calling it with `outerCallback` made such callback be called twice (by `end` and `close`).